ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/RedundantAttributeSuffixAnalyzer.fs

```diff
@@ -0,0 +1,22 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Analyzers
+
+open JetBrains.ReSharper.Feature.Services.Daemon
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi
+
+[<ElementProblemAnalyzer(typeof<IAttribute>,
+                         HighlightingTypes = [| typeof<RedundantAttributeSuffixWarning> |])>]
+type RedundantAttributeSuffixAnalyzer() =
+    inherit ElementProblemAnalyzer<IAttribute>()
+
+    override x.Run(attribute, _, consumer) =
+        let attributeName = attribute.ReferenceName.ShortName
+        if not (attributeName.EndsWith("Attribute")) then () else
```

Comment:
Please also check that the whole name is not `Attribute`.

Reply:
Another note: please use `CommonUtil.endsWith` or pass `StringComparison.Ordinal` to the `EndWith` method. You're doing a culture-dependent check here which is wrong here (since it might produce different results in some cultures) and is slower than using `StringComparison.Ordinal`.